### PR TITLE
[WIP] Bootstrap with nuget

### PR DESCRIPTION
**Pull request by anurse**
We're close, but not quite there, but I want to get this up and have Jenkins try building it. This changes dotnet cli to bootstrap with NuGet.

### CODE REVIEW

**Comment by anurse** [comment]
Something's wrong with the PATH I'm setting in `build.sh` so CentOS (at least) isn't finding the dotnet stage0. @brthor @piotrpMSFT @Sridhar-MS I have to duck out for a bit, if one of you could take a brief look, we might be able to get this in soon! :)

**Comment by anurse** [comment]
Also it looks like maybe `ilc` didn't get published properly?

**Comment by Sridhar-MS** [comment]
@anurse should this PR be opened against rel/1.0.0 instead of master?

File: scripts/build/build.sh [hunk 1]
```diff
@@ -44,8 +44,9 @@ if [ ! -z "$OFFLINE" ]; then
     info "Skipping Tools and Package Download: Offline build"
 else
    $REPOROOT/scripts/obtain/install-tools.sh
-    
-   $REPOROOT/scripts/build/restore-packages.sh
+
+   # Restore using the stage 0
+   PATH="$REPOROOT/artifacts/$RID/stage0/bin:$PATH" $REPOROOT/scripts/build/restore-packages.sh
```
Comment (brthor) [inline scripts/build/build.sh]: Taking  a look in some previous builds I see stage0 in this path:

`$REPOROOT/.dotnet_stage0 / ubuntu.14.04-x64 / bin`

Comment (Sridhar-MS) [inline scripts/build/build.sh]: Yes. Thats the correct location.

Comment (Sridhar-MS) [inline scripts/build/build.sh]: I am adding a commit to `anurse/bootstrap-with-nuget-finally` to fix this

**Comment by Sridhar-MS** [comment]
@anurse since you have opened the PR against master, the stage0 is dowloaded from the `dev` channel instead of the `beta` channel. Please reopen this PR against rel/1.0.0

**Comment by anurse** [comment]
Despite the PR target, the branch is based on `rel/1.0.0`. I'll re-open the PR though.

File: scripts/build/build.sh (hunk 1 above)
Comment (anurse) [inline scripts/build/build.sh]: Thanks!

**Pull request closed without merging**